src/components/instruction-guide/instruction-guide.tsx

```diff
@@ -0,0 +1,128 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import QuestionSquaredIcon from '@/components/icons/question-squared';
+import VideoSquaredIcon from '@/components/icons/video-squared';
+import { useToggle } from '@react-hookz/web';
+
+import CloseIcon from '@mui/icons-material/Close';
+import { Button, Stack, Typography, IconButton, Card } from '@mui/material';
+
+import SideBarVideoInstruction from './side-bar-video';
+
+type Props = {
+  title: string;
+  desc: string;
+  btnLink: string;
+  btnText: string;
+  videoUrl: string;
+  onSideDialogClose?: () => void;
+};
+
+export function InstructionGuide({
+  title,
+  desc,
+  btnLink,
+  btnText,
+  videoUrl,
+  onSideDialogClose,
+}: Props) {
+  const [open, setOpen] = useState(false);
+  const [openVideoPlayer, toggleVideoPlayer] = useToggle(false);
+
+  let hasSeenDialog: { [key: string]: boolean } | null;
+
+  useEffect(() => {
+    hasSeenDialog = JSON.parse(
+      localStorage.getItem('coach-mark-guide') || '{}'
+    );
+  }, []);
+
+  useEffect(() => {
+    if (hasSeenDialog && !hasSeenDialog.hasOwnProperty(title)) {
+      setOpen(true);
+    }
+  }, []);
+
+  const handleClick = () => {
+    const updatedDialog = { ...hasSeenDialog, [title]: true };
+
+    localStorage.setItem('coach-mark-guide', JSON.stringify(updatedDialog));
+    setOpen(false);
+  };
```

Comment:
We have the `useLocalStorageValue` hook on `react-hookz` library (https://react-hookz.github.io/web/?path=/docs/side-effect-uselocalstoragevalue--example)

You're using the useEffect for initializing the value of the `open` state and the `hasSeenDialog` , but since the `hasSeenDialog` variable isn't a state, this change will happen outside of the rendering tree

I would recommend you get the value of the localstorage before initializing everything else, so you can set the values of the state based on it


```suggestion
import { useLocalStorageValue } from '@react-hookz/web';

//....
  const {value: coachMark, set: setCoachMark} = useLocalStorageValue('coach-mark-guide'); // it'll automatically parse the value
  const open = !coachMark?.[title]; // I would rename it to `hasSeen`
  const [openVideoPlayer, toggleVideoPlayer] = useToggle(false);

  const handleClick = () => {
    setCoachMark({
    	...coachMark,
    	[title]: true
    });
    // it'll automaticaly update the `coachMark` and the `open` variables
  };
```